text/0149-exponential-histogram.md

```diff
@@ -0,0 +1,162 @@
+# Add exponential bucketing to histogram protobuf
+
+Add exponential bucketing to histogram protobuf
+
+## Motivation
+
+Currently, the OTEL protobuf [protocol](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto) only supports explicit bound buckets. Each bucket's bound and count must be explicitly defined. This is inefficient to transport buckets whose bounds have a pattern, such as exponential (ie. log scale) buckets. More importantly, without bucket pattern info, the receiver may not be able to optimize its processing on these buckets. With a protocol that also supports exponential buckets, the bounds can be encoded with just a few parameters, regardless of the number of buckets, and the receiver can optimize processing based on the knowledge that these buckets are exponential.
+
+The explicit bucket type will be kept, as a fallback for arbitrary bucket bounds. For example, Prometheus histograms often come with arbitrary user defined bounds.
+
+Exponential buckets are chosen to be added because they are very good at representing [long tail](https://en.wikipedia.org/wiki/Long_tail) distributions, which are common in the OTEL target application like response time measurement. Exponential buckets (ie. log scale buckets) need far fewer buckets than linear scale buckets to cover the wide range of a long tail distribution. Furthermore, percentiles/quantiles can be computed from exponential buckets with constant relative error across the full range.
+
+## Explanation
+
+Exponential buckets will be added. In general, bucket bounds are in the form of:
+
+```
+bound = base ^ exponent
+```
+
+where base is a parameter of a bound series, and exponent is an integer. Note that exponent may be positive, negative, or 0. Such bounds are also commonly known as "log scale bounds".
+
+## Internal details
+
+Proposed message type to be added:
+
+```
+message ExponentialBuckets {
+    double base = 1;
```

Comment:
typo? `base=1` will result in `bound = 1^ exponent`, which is always `==1`

Reply:
this is a protocol buffer definition, 1 = the encoded field number for serialization no the value of the base.

Reply:
whops! Thx for the clarification.